easybuild/easyblocks/c/clang.py

```diff
@@ -113,6 +124,91 @@ def __init__(self, *args, **kwargs):
+
+        # Be forgiving if someone places a runtime under projects since it is pretty new
+        for project in [p for p in self.cfg['llvm_projects'] if p in KNOWN_LLVM_RUNTIMES]:
+            msg = "LLVM project %s included but this should be a runtime, moving to runtime list!" % project
+            self.log.warning(msg)
+            self.cfg.update('llvm_runtimes', [project], allow_duplicate=False)
+            # No cleaner way to remove an element from the list
+            self.cfg['llvm_projects'] = [p for p in self.cfg['llvm_projects'] if p != project]
+            print_warning(msg)
+
+        for project in [p for p in self.cfg['llvm_projects'] if p not in KNOWN_LLVM_PROJECTS]:
+            msg = "LLVM project %s included but not recognised, this project will NOT be sanity checked!" % project
+            self.log.warning(msg)
+            print_warning(msg)
+
+        for runtime in [r for r in self.cfg['llvm_runtimes'] if r not in KNOWN_LLVM_RUNTIMES]:
+            msg = "LLVM runtime %s included but not recognised, this runtime will NOT be sanity checked!" % runtime
+            self.log.warning(msg)
+            print_warning(msg)
+
+        # keep compatibility between using llvm_projects/llvm_runtimes vs using flags
+        if LooseVersion(self.version) >= LooseVersion('14'):
+            self.cfg.update('llvm_projects', ['llvm', 'clang'], allow_duplicate=False)
+            self.cfg.update('llvm_runtimes', ['compiler-rt', 'openmp'], allow_duplicate=False)
+            if self.cfg['usepolly']:
+                self.cfg.update('llvm_projects', 'polly', allow_duplicate=False)
+            if self.cfg['build_lld']:
+                self.cfg.update('llvm_projects', ['lld'], allow_duplicate=False)
+                self.cfg.update('llvm_runtimes', ['libunwind'], allow_duplicate=False)
+            if self.cfg['build_lldb']:
+                self.cfg.update('llvm_projects', 'lldb', allow_duplicate=False)
+            if self.cfg['libcxx']:
+                self.cfg.update('llvm_runtimes', ['libcxx', 'libcxxabi'], allow_duplicate=False)
+            if self.cfg['build_extra_clang_tools']:
+                self.cfg.update('llvm_projects', 'clang-tools-extra', allow_duplicate=False)
+
+        # ensure libunwind is there if lld is there
+        if 'lld' in self.cfg['llvm_projects']:
+            self.cfg.update('llvm_runtimes', 'libunwind', allow_duplicate=False)
+
+        # ensure libcxxabi is there if libcxx is there
+        if 'libcxx' in self.cfg['llvm_runtimes']:
+            self.cfg.update('llvm_runtimes', 'libcxxabi', allow_duplicate=False)
+
+        build_targets = self.cfg['build_targets']
+        # define build_targets if not set
+        if build_targets is None:
+            arch = get_cpu_architecture()
+            try:
+                default_targets = DEFAULT_TARGETS_MAP[arch][:]
+                # If CUDA is included as a dep, add NVPTX as a target
+                if get_software_root("CUDA"):
```

Comment:
Doing this in the constructor doesn't work, since the modules for the dependencies are only loading in `prepare_step` => https://github.com/easybuilders/easybuild-easyblocks/issues/2871

Reply:
I guess the key is in the comment here, this should be like what is done in LAMMPS (https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/l/lammps.py#L175-L177)

Reply:
@ocaisa please mention that in #2871